src/OrchardCore.Modules/OrchardCore.Autoroute/Indexes/AutoroutePartIndex.cs

```diff
@@ -107,13 +107,11 @@ public override async Task UpdatedAsync(UpdateContentContext context)
         public void Describe(DescribeContext<ContentItem> context)
         {
             context.For<AutoroutePartIndex>()
+                .When(contentItem => contentItem.Has<AutoroutePart>())
```

Comment:
The second problem with using `When` here is it will stop the `.Map` being called, because the part no longer has the `AutoroutePart` because it has been removed previously by the handler, so will stop the delete of the item.

But in this case I think we could check in `When` to see if `.Has<AutoroutePart>() || _part.Removed.Contains()`, and return true, which would allow the `.Map` to continue, and do what it needs to, i.e issue the necessary deletes.

Reply:
But it is in the item you enlisted in the `_partRemoved`

It's a first pass, and i saw another problem, i will see tomorrow

I thought about using `_part.Removed` but i was thinking of not adding a bad closure, but that's the same in the map delegate, so maybe

I also need to check more in depth what return `Has<>` and `As<>`

Reply:
Okay, just saw that you remove the part in the handler section of the index provider ;)

And i was planning to do it in the map delegate, as alias does for now

Also, the record added when removing the part needs to be kept in the index even if we publish again the item, otherwise another running instance may be not in sync, will see this night / tomorrow